### ISSUE TITLE
build failure

### ISSUE THREAD

**Issue by 0323pin** [conformal/spectrwm#534]
3 days after release, an updated package seems to be only available on OpenBSD.
I did try on the release day and building failed.

Now, I've applied the OpenBSD patches but, still ...
building 3.5.0 on NetBSD fails with the following errors:

```
===> Configuring for spectrwm-3.5.0
=> Checking for portability problems in extracted files
===> Building for spectrwm-3.5.0
--- spectrwm.c ---
--- swm_hack.c ---
--- spectrwm.c ---
ln -sf ../spectrwm.c
--- swm_hack.c ---
ln -sf ../lib/swm_hack.c
--- spectrwm.c ---
ln -sf ../version.h
--- swm_hack.so ---
--- spectrwm.c ---
ln -sf ../linux/queue_compat.h
--- spectrwm.o ---
--- swm_hack.so ---
gcc -pipe -O2 -I/usr/pkg/include -I/usr/X11R7/include -I/usr/include -I/usr/pkg/include/freetype2 -std=c99 -Wmissing-prototypes -Wall -Wextra -Wshadow -Wno-uninitialized -g -I. -I/usr/pkg/include -I/usr/pkg/include/freetype2 -DSWM_LIB=\"/usr/pkg/lib/libswmhack.so.0.0\" -c -shared -fpic -DPIC -o swm_hack.so swm_hack.c
--- spectrwm.o ---
gcc -pipe -O2 -I/usr/pkg/include -I/usr/X11R7/include -I/usr/include -I/usr/pkg/include/freetype2 -std=c99 -Wmissing-prototypes -Wall -Wextra -Wshadow -Wno-uninitialized -g -I. -I/usr/pkg/include -I/usr/pkg/include/freetype2 -DSWM_LIB=\"/usr/pkg/lib/libswmhack.so.0.0\"  -D_OPENBSD_SOURCE -I/usr/pkg/include -I/usr/X11R7/include -I/usr/include -I/usr/pkg/include/freetype2 -c spectrwm.c -o spectrwm.o
--- libswmhack.so.0.0 ---
gcc -L/usr/pkg/lib -Wl,-R/usr/pkg/lib -Wl,-zrelro -L/usr/X11R7/lib -Wl,-R/usr/X11R7/lib -L/usr/lib -Wl,-R/usr/lib -shared -fpic -o libswmhack.so.0.0 swm_hack.so
--- spectrwm.o ---
In file included from /usr/include/stdio.h:595,
                 from spectrwm.c:56:
spectrwm.c: In function 'scan_config':
spectrwm.c:16733:14: error: 'SWM_CONF_FILE' undeclared (first use in this function); did you mean 'SWM_COUNT_TILED'?
16733 |         ret, SWM_CONF_FILE);
      |              ^~~~~~~~~~~~~
spectrwm.c:16733:14: note: each undeclared identifier is reported only once for each function it appears in
spectrwm.c:16781:8: error: 'SWM_CONF_FILE_OLD' undeclared (first use in this function); did you mean 'SWM_COUNT_TILED'?
16781 |        SWM_CONF_FILE_OLD);
      |        ^~~~~~~~~~~~~~~~~
spectrwm.c: At top level:
spectrwm.c:1521:14: warning: 'focusout' used but never defined
 1521 | static void  focusout(xcb_focus_out_event_t *);
      |              ^~~~~~~~
spectrwm.c:1537:20: warning: 'get_gravity_label' declared 'static' but never defined [-Wunused-function]
 1537 | static const char *get_gravity_label(uint8_t);
      |                    ^~~~~~~~~~~~~~~~~
spectrwm.c:1739:14: warning: 'update_gravity' used but never defined
 1739 | static void  update_gravity(struct ws_win *);
      |              ^~~~~~~~~~~~~~
*** [spectrwm.o] Error code 1

make: stopped in /usr/pkgsrc/wip/spectrwm/work/spectrwm-SPECTRWM_3_5_0/netbsd
1 error

make: stopped in /usr/pkgsrc/wip/spectrwm/work/spectrwm-SPECTRWM_3_5_0/netbsd
*** Error code 2

Stop.
```

Hints?
TIA

**Comment by LordReg** [comment]
The NetBSD build issues should be fixed in master. Also added NetBSD to the XInput 2 whitelist. Please verify it works for you and we will roll a patch release soon.

**Comment by 0323pin** [comment]
@LordReg Thanks! I can confirm it builds from master.
I'm away for one week with limited WiFi access but, please ping me when a patched release is out and I'll try to push the update. Else, I'm back home next Friday and can do it then.

**Comment by 0323pin** [comment]
Would you prefer if I apply https://github.com/conformal/spectrwm/commit/624b67f32723ab600cbc272e9199cadd66659825 as a patch  and update our package? Or, is a point release arround the corner?

Just let me know.

**Comment by 0323pin** [comment]
@LordReg Take your time with the new point release, I've applied the relevant changes as a patch and bumped the NetBSD package to 3.5.0, commit log for reference: http://mail-index.netbsd.org/pkgsrc-changes/2023/11/03/msg285782.html

**Comment by LordReg** [comment]
Thanks for going ahead with the patched 3.5.0. Held off to get some additional fixes in: https://github.com/conformal/spectrwm/releases/tag/SPECTRWM_3_5_1

**Comment by 0323pin** [comment]
@LordReg thanks! https://mail-index.netbsd.org/pkgsrc-changes/2023/11/25/msg287655.html
